File: data structures/stack/javascript/stack.js
// Stack structure

class Stack {
    constructor() {
        this.data = [];
    }
    
    push(item) {
        this.data.push(item);
    }

    pop() {
        return this.data.pop();
    }

    isEmpty() {
        return this.data.length == 0;
    }
}

// Tests

let stack = new Stack();
console.assert(stack.isEmpty(), "Stack is empty");
stack.push("First");
stack.push("Second");
stack.push("Third");
console.assert(stack.pop(), "Popped element Third");
stack.pop();
console.assert(stack.isEmpty() == false, "Stack isn't empty");

File: data structures/stack/swift/stack.swift
// Stack structure

class Stack<T> {
    private var data = [T]()
    
    func push(element: T) {
        data.append(element)
    }
    
    func pop() -> T? {
        return data.popLast()
    }
    
    var isEmpty: Bool {
        return data.isEmpty
    }
    
}

// Tests

let stack = Stack<String>()
assert(stack.isEmpty, "Stack is empty")
stack.push(element: "First")
stack.push(element: "Second")
stack.push(element: "Third")
assert(stack.pop() == "Third", "Popped element Third")
_ = stack.pop()
assert(stack.isEmpty == false, "Stack isn't empty")
